libc/src/impl.c

```diff
@@ -0,0 +1,1193 @@
+//
+// If you have only one single C file, you can include std header files in it,
+// and then done. It automatically includes impl.c(this file). It is compatible
+// to a lot of old CKB code.
+//
+// If you have more than one C file, and it's a big project. You need to define
+// macro CKB_DECLARATION_ONLY for every C files. Compile very C file, together
+// with this impl.c (compiled *without* macro CKB_DECLARATION_ONLY) and then
+// link them together.
+//
+
+#ifndef __IMPL_INCLUDED__
```

Comment:
Let's add a `CKB_` prefix to make it `__CKB_IMPL_INCLUDED__`, so as to avoid unnecessary conflicts.